src/client/Microsoft.Identity.Client/WwwAuthenticateParameters.cs

```diff
@@ -231,26 +462,12 @@ internal static WwwAuthenticateParameters CreateWwwAuthenticateParameters(IDicti
                 wwwAuthenticateParameters.Error = value;
             }
 
-            return wwwAuthenticateParameters;
-        }
-
-        /// <summary>
-        /// Extracts a key value pair from an expression of the form a=b
-        /// </summary>
-        /// <param name="assignment">assignment</param>
-        /// <returns>Key Value pair</returns>
-        private static KeyValuePair<string, string> ExtractKeyValuePair(string assignment)
-        {
-            string[] segments = CoreHelpers.SplitWithQuotes(assignment, '=')
-                .Select(s => s.Trim().Trim('"'))
-                .ToArray();
-
-            if (segments.Length != 2)
+            if (values.TryGetValue("nonce", out value))
```

Comment:
Do we have test cases where the parameter is not lowercase, for ex:
`        [DataRow("AuthScheme", "realm=someRealm Nonce=nonce, ERROR=\"err\"")]`